### PR TITLE
Automatic publishing of documentation

### PR DESCRIPTION
**Pull request by bettinaheim**
Our documentation is checked in on a branch in this repository. The idea is that each documentation version is a subfolder on that branch, and a redirect at the root is pointing to a specific version. 

This PR set up navigation between different version by adding an entry titled `Other Versions` to the index, and the corresponding page contains links to download each version and to the corresponding docs.

The checked in documentation can be deployed to nvidia.github.io/cuda-quantum by either configuring the repo to do so whenever that branch is updated (currently the case), or by setting up a custom workflow for deployment. I didn't see a straightforward way to update only a specific folder when deploying a docs update, so it seems to be the most reasonable solution to stick with the current setting and push incremental updates to the branch where all docs are checked in.

This PR adds a workflow that is triggered when the package deployment to GHCR completes after updates to main and releases branches.
This workflow will check out the current docs version, determine which folder needs to be update, replace that folder with the docs artifact produced by the triggering workflow, and push the update to the docs branch. Right now, there is no staging set up, and the update is pushed directly to the live branch.

### CODE REVIEW

File: Building.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ To learn more about how to use CUDA Quantum, please take a look at the [CUDA
 Quantum Documentation][cuda_quantum_docs]. The page also contains [installation
 instructions][official_install] for released packages.
 
-[cuda_quantum_docs]: https://nvidia.github.io/cuda-quantum/
+[cuda_quantum_docs]: https://nvidia.github.io/cuda-quantum/latest
```
Comment (bettinaheim) [inline Building.md:7]: I think it may make sense to direct the nvidia.github.io/cuda-quantum link to the docs for the latest stable release on NGC opposed to the docs for the latest main. However, if someone is looking at the GitHub source code, the docs version they want is latest. I am hence explicitly setting the link here to that version.

File: .github/workflows/deployments.yml [hunk 2]
```diff
@@ -1,5 +1,11 @@
 on:
   workflow_dispatch:
+    inputs:
+      update_registry_cache:
```
Comment (bettinaheim) [inline .github/workflows/deployments.yml:4]: This was an oversight - there was no way to manually push a build cache update to the registry. I randomly added it here but will make a separate PR if desired.

**Review by amccaskey — APPROVED** [review]

File: .github/workflows/build_packages.yml [hunk 3]
```diff
@@ -154,9 +179,11 @@ jobs:
       - name: Build documentation
         id: docs_build
         run: |
+          docs_version="CUDA_QUANTUM_VERSION=${{ needs.metadata.outputs.image_tag }}"
+
           docker load --input ${{ needs.release_build.outputs.cudaqdev_tarfile }}
           docker run --rm -dit --name cuda-quantum-dev cuda-quantum-dev:local
-          (docker exec cuda-quantum-dev bash scripts/build_docs.sh && built=true) || built=false
+          (docker exec cuda-quantum-dev export $docs_version && bash scripts/build_docs.sh && built=true) || built=false
```
Comment (bettinaheim) [inline .github/workflows/build_packages.yml:186]:
```suggestion
          (docker exec cuda-quantum-dev bash -c "export $docs_version && bash scripts/build_docs.sh" && built=true) || built=false
```
